pkg/jtracer/jtracer.go

```diff
@@ -14,16 +14,98 @@
 
 package jtracer
 
-import "github.com/opentracing/opentracing-go"
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel"
+	otbridge "go.opentelemetry.io/otel/bridge/opentracing"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.opentelemetry.io/otel/trace"
+)
 
 type JTracer struct {
-	OT opentracing.Tracer
+	OT     opentracing.Tracer
+	OTEL   trace.TracerProvider
+	closer func(ctx context.Context) error
 }
 
-func OT(t opentracing.Tracer) JTracer {
-	return JTracer{OT: t}
+var once sync.Once
+
+func New(serviceName string) (*JTracer, error) {
+	ctx := context.Background()
+	tracerProvider, err := initOTEL(ctx, serviceName)
+	if err != nil {
+		return nil, err
+	}
+	// Use the bridgeTracer as your OpenTracing tracer(otTrace).
+	otelTracer := tracerProvider.Tracer("github.com/jaegertracing/jaeger/pkg/jtracer")
+	otTracer, wrappedTracerProvider := otbridge.NewTracerPair(otelTracer)
```

Comment:
does this provide Shutdown function? If it does, does it also invoke tracerProvider.Shutdown?

Reply:
I don't see. Sorry!